javagen/src/main/java/com/azure/autorest/template/PatchModelTemplate.java

```diff
@@ -0,0 +1,290 @@
+
+                JavaVisibility methodVisibility = property.getClientFlatten() ? JavaVisibility.Private : JavaVisibility.Public;
+
+                // setter
+                if (!property.getIsReadOnly() && !(settings.isRequiredFieldsAsConstructorArgs() && property.isRequired()) && methodVisibility == JavaVisibility.Public) {
+                    generateSetterJavadoc(classBlock, model, property);
+                    TemplateUtil.addJsonSetter(classBlock, settings, property.getSerializedName());
+                    IType setterPropertyType = ClientModelUtil.isClientModel(property.getWireType()) ? new ClassType.Builder().packageName(((ClassType) property.getWireType()).getPackage()).name(((ClassType) property.getWireType()).getName() + "Patch").build() : property.getWireType();
+                    classBlock.method(methodVisibility, null, String.format("%s %s(%s %s)",
+                                    patchClassNameWithBaseType, property.getSetterName(), setterPropertyType, property.getName()),
+                            (methodBlock) -> {
+                                String expression;
+                                if (propertyClientType.equals(ArrayType.ByteArray)) {
+                                    expression = String.format("CoreUtils.clone(%s)", property.getName());
+                                } else {
+                                    expression = property.getName();
+                                }
+                                methodBlock.ifBlock(String.format("%s == null", property.getName()),
+                                                (ifBlock) -> ifBlock.line("this.%s = Option.empty();", property.getName()))
+                                        .elseBlock((elseBlock) -> {
+                                            String propertyConversion = propertyType.convertFromClientType(expression);
+                                            elseBlock.line("this.%s = Option.of(%s);", property.getName(), propertyConversion);
+                                        });
+                                methodBlock.methodReturn("this");
+                            });
+                }
+
+                if (property.isAdditionalProperties()) {
+                    classBlock.annotation("JsonAnySetter");
+                    MapType mapType = (MapType) property.getClientType();
+                    classBlock.packagePrivateMethod(String.format("void %s(String key, %s value)", property.getSetterName(), mapType.getValueType()), (methodBlock) -> {
+                        methodBlock.ifBlock(String.format("%s == null", property.getName()), ifBlock ->
+                                ifBlock.line("%s = new HashMap<>();", property.getName()));
+                        methodBlock.line("%s.put(%s, value);", property.getName(), model.getNeedsFlatten() ? "key.replace(\"\\\\.\", \".\")" : "key");
+                    });
+                }
```

Comment:
We might ignore the `additionalProperties` for now. You will need proper test if you enable this.